Modules/xASL_module_ASL.m

```diff
@@ -101,14 +102,40 @@
 		x.Output_im = rmfield(x.Output_im,'ASL');
 	end
 
-	if isfield(x,'Output') && isfield(x.Output,'ASL')
-		x.Output = rmfield(x.Output,'ASL');
+	if isfield(x, 'Output') && isfield(x.Output, 'ASL') && isfield(x.Output.ASL, x.SESSION)
+		x.Output = rmfield(x.Output.ASL, x.SESSION);
 	end
 	
 	% And saved the cleaned up version
 	xASL_adm_SaveX(x);
+	
+	% If we rerun completely, then we do not the reload QC for that session, otherwise we load it from QC.json if present.
+	bCompleteRerun = true;
+else
+	x = xASL_adm_LoadX(x, [], true); % assume x.mat is newer than x
+end
+
+% Load QC.json and save it to x-struct
+QC_Path = fullfile(x.D.ROOT, x.SUBJECT, ['QC_collection_' x.SUBJECT '.json']);
```

Comment:
What you write here is needed every time you load the QC_collection JSON right, not when you loadX or anything else?

Then I would be in favor to move this to a separate function that deals with loading QC_Collection*.json (if it needs to be the same every time, irrespective of where we run it).

If this is something that we only need at the start of the ASL module, then we can leave it here.

BTW: why can't this loop go to xASL_qc_RemoveOutdatedQC?